torch/lib/c10d/TCPStore.cpp

```diff
@@ -522,6 +726,28 @@ bool TCPStore::deleteKey(const std::string& key) {
   return (numDeleted == 1);
 }
 
+void TCPStore::watchKey(const std::string& key, WatchKeyCallback callback) {
+  // Only allow one thread to perform watchKey() at a time
+  const std::lock_guard<std::mutex> watchKeyLock(watchKeyMutex);
+
+  // Register callback with TCPStoreMasterDaemon to call TCPStoreWorkerDaemon on
+  // key change
+  std::string regKey = regularPrefix_ + key;
+  tcpStoreWorkerDaemon_->addCallback(regKey, callback);
+  tcputil::sendValue<QueryType>(listenSocket_, QueryType::WATCH_KEY);
+  tcputil::sendString(listenSocket_, regKey);
+
+  // Block until callback has been registered successfully
+  std::unique_lock<std::mutex> callbackRegistrationLock(
+      callbackRegistrationMutex);
+  tcpStoreWorkerDaemon_->getCallbackRegisteredCondVar().wait(
+      callbackRegistrationLock,
+      [&] { return tcpStoreWorkerDaemon_->getCallbackRegisteredData(); });
+
+  // Reset payload for next callback
+  tcpStoreWorkerDaemon_->setCallbackRegisteredData(false);
```

Comment:
sorry for misleading you. My original intention is to wrap these two lines into one function in `tcpStoreWorkerDaemon_`, so that we don't need to expose its internal CVs.

Reply:
Gotcha, made the change!